src/ShoppingFeedHelper.php

```diff
@@ -151,7 +151,19 @@ public static function get_sf_account_options() {
 	 * @return array
 	 */
 	public static function get_sf_yoast_options() {
-		$yoast_options = get_option( Options::SF_YOAST_OPTIONS, [] );
+
+		if ( false === get_option( Options::SF_YOAST_OPTIONS ) )
```

Comment:
The current function `get_sf_yoast_options` seem fine to me, I don't see the point or rewriting it. The only change I could see if changing the default value to `'1'` to activate Yoast primary category support if available.

```
$yoast_options = get_option( Options::SF_YOAST_OPTIONS, [] );

return wp_parse_args(
	$yoast_options,
	[
		'use_principal_categories' => '1', <------
	]
);
```